### PR TITLE
Token

### PR DESCRIPTION
**Pull request by AlexMegas**
[1]
PATCH /api/books/5e85a0e1c218b46014ac3e03 500 91.021 ms - 814
TypeError: Cannot read property 'headers' of undefined
--> can not understand how to return (user) from checkToken to checkIfOwner

[2]
../controllers/book
updateBookById => response is not updated book, why?

### CODE REVIEW

File: controllers/book.js [hunk 1]
```diff
@@ -66,21 +82,30 @@ exports.getBookByQuery = (req, res) => {
...
+// find book by id and delete
 exports.deleteBookById = (req, res) => {
-    const bookId = req.params.id;// find book by id and delete
-    Book.findOneAndDelete({ _id: bookId })
-        .then(res.send(`Record with id:${bookId} --> deleted from DB!`))
-        .catch(err => {
-            console.log(err);
-            res.status(500).send();
+    const bookId = req.params.id;
+    Book.findOneAndDelete({
```
Comment (IBurychka) [inline controllers/book.js:102]: Also you can use method findByIdAndDelete

File: middlewares/auth.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        })
+        .catch(err => res.status(500).json({
+            message: 'Server error'
+        }));
+};
+
+
+const checkIfOwner = (req, res, next) => {
+    const bookId = req.params.id;
+    checkToken()
```
Comment (IBurychka) [inline middlewares/auth.js:32]: remove checkToken() you need to chek token in route using middleWare

File: controllers/book.js (hunk 1 above)
Comment (AlexMegas) [reply controllers/book.js:102]: modified from "findOneAndDelete" into "findByIdAndDelete"

**Comment by AlexMegas** [comment]
closed due to upcoming changes.

**Pull request closed without merging**